zhmccli/_cmd_cpc.py

```diff
@@ -994,31 +1006,83 @@ def cmd_dpm_import(cmd_ctx, cpc_name, options):
     else:
         mapping_obj = None
 
-    config_dict['preserve-uris'] = bool(preserve_uris)
-    config_dict['preserve-wwpns'] = bool(preserve_wwpns)
-    if mapping_obj:
-        config_dict['adapter-mapping'] = \
-            convert_adapter_mapping(cmd_ctx, mapping_obj)
-    else:
-        if 'adapter-mapping' in config_dict:
-            del config_dict['adapter-mapping']
+    _log_mapping_and_overwrite_if_needed(cmd_ctx, config_dict, mapping_obj,
+                                         dpm_file, mapping_file)
+    _dump_config(config_dict)
+    cmd_ctx.spinner.stop()
+
+    if options["yes"]:
+        click.echo("already confirmed!")
+
+    message = "Are you sure you want to import the DPM configuration "
+    ("from {f} into CPC {c}, replacing its current DPM configuration?"
+     .format(c=cpc_name, f=dpm_file))
+
+    if not click.confirm(message):
```

Comment:
Does the --yes flag still work with this explicit confirmation?

Reply:
with the updates I made, --yes